File: SchoolWebApp-API/Project.Core/Entities/CBE/Assessments/SpecificOutcome.cs
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Assessments
{
    public class SpecificOutcome: SettingsBase
    {
        public int SubStrandId { get; set; }
        public SubStrand? SubStrand { get; set; } = null;

        public int LearningLevelId { get; set; }
        public LearningLevel? LearningLevel { get; set; } = null;

        public int BroadOutcomeId { get; set; }
        public BroadOutcome? BroadOutcome { get; set; } = null;

        public int GeneralOutcomeId { get; set; }
        public GeneralOutcome? GeneralOutcome { get; set; } = null;

        public List<Competency> Competencies { get; set; } = new();
        public List<StudentAssessment> StudentAssessments { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/cbe/assessments/models/specific-outcome.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class SpecificOutcome extends ResourceModel<SpecificOutcome> {
    public name?: string;
    public description?: string;
    public rank?: number;
    public subStrandId?: number;
    public subStrand?: any;
    public sessionId?: number;
    public session?: any;
    public broadOutcomeId?: number;
    public broadOutcome?: any;
    public generalOutcomeId?: number;
    public generalOutcome?: any;

    // Used by add/edit form for cascading filters
    public academicYearId?: number;
    public curriculumId?: number;
    public learningLevelId?: number;
    public subjectId?: number;
    public strandId?: number;

    constructor(model?: Partial<SpecificOutcome>) {
        super(model);
    }
}
